File: oshibana-web/src/lib/scryfall/prices.ts
import type { URI } from "./card";

export interface Prices {
    usd?: string;
    usd_foil?: string;
    usd_etched?: string;
    eur?: string;
    eur_foil?: string;
    eur_etched?: string;
    tix?: string;   
}

export interface PurchaseUris {
    tcgplayer?: URI;
    cardmarket?: URI;
    cardhoarder?: URI;
}

File: schemas/src/scryfall/card/prices.rs
use crate::generate_record_builder_and_dt;
use serde::{Deserialize, Serialize};

generate_record_builder_and_dt! {
    #[derive(Debug, Serialize, Deserialize)]
    Prices {
        usd: Option<String>,
        usd_foil: Option<String>,
        usd_etched: Option<String>,
        eur: Option<String>,
        eur_foil: Option<String>,
        tix: Option<String>,
    }
}
